tickit/devices/eiger/eiger.py

```diff
@@ -0,0 +1,106 @@
+from typing import Awaitable, Callable, Optional
+
+from aiohttp import web
+
+from tickit.adapters.httpadapter import HTTPAdapter
+from tickit.adapters.interpreters.endpoints.http_endpoint import HTTPEndpoint
+from tickit.adapters.servers.http_server import HTTPServer
+from tickit.core.adapter import ConfigurableAdapter
+from tickit.core.device import ConfigurableDevice, DeviceUpdate
+from tickit.core.typedefs import SimTime
+from tickit.devices.eiger.eiger_config import EigerConfig
+from tickit.utils.byte_format import ByteFormat
+from tickit.utils.compat.typing_compat import TypedDict
+
+
+class Eiger(ConfigurableDevice):
+    """A device class for the Eiger detector.
+
+    ...
+    """
+
+    Inputs: TypedDict = TypedDict("Inputs", {"foo": bool})
+
+    Outputs: TypedDict = TypedDict("Outputs", {"bar": float})
+
+    config: EigerConfig
```

Comment:
We shouldn't reference the config. The existence of an auto-generated config class is an implementation detail of our (de)serialization system and is subject to change

Reply:
in the `__init__()` i also do 
```python
self.config = EigerConfig()
```

Reply:
You shouldn't. The config parameters will be passed to the `__init__` of the device. (Configs are auto-generated by introspecting the signature of the constructor)

Reply:
Ah, wait, we seem to be chatting about two different `EigerConfig`s, the way you're passing yours is fine, may be worth changing the name though (probably worth having a look at [this decorator](https://github.com/dls-controls/tickit/blob/c16829a6188e163bcc571e6b75a1f203b642b13d/tickit/utils/configuration/configurable.py#L199).)

Reply:
The Eiger config has many values, surely there is scope for structuring and organising config i.e.
```yaml
my_device:
  foo: a
  bar:
    baz: 2
    something_else: 4
```
And surely that would be achieved by passing dataclasses, which apischema loves, to `__init__`?

Reply:
Yeah there is an overloading of the word "config" going on here. The eiger manual calls these values its "config". Maybe rename this one to `EigerLiveConfig`? Opinions @garryod?

Reply:
Reckon it's probably best to leave the `Config` prefix for tickit configuration - though things may change with #4 / #16. Could possibly go with `EigerSettings`.

Reply:
Renamed to `EigerSettings` then